### ISSUE TITLE
View.getZoom return 0 if use view.scale(+1, -1)

### ISSUE THREAD

**Issue by chrisDupuis** [paperjs/paper.js#1411]
# Description/Steps to reproduce
I use view.scale(+1, -1) in order to have a X, and Y axis going up (direct axis system)
The problem is when doing that, the method view.getZoom return 0
because the code is 
`return scaling ? (scaling.x +scaling.y) / 2 : 0;`

Why not changing this line with:
`return scaling ? (Math.abs(scaling.x) + Math.abs(scaling.y)) / 2 : 0;`

Is there  an other way to have a view with Y axis going up

# Link to reproduction test-case

# Expected result

# Additional information

**Comment by sapics** [comment]
@sasensi How about this?
```js
return scaling ?
    (scaling.x * scaling.y < 0 ? Math.abs(scaling.x) + Math.abs(scaling.y) : scaling.x + scaling.y) / 2 : 0
```

**Comment by lehni** [comment]
I think @sapics proposal sounds good! Would that work in all combinations of x and y?

**Comment by sapics** [comment]
@lehni @sasensi  It makes error in x=0 and y=0 or scaling is null, but it looks not important, because `zero zoom` looks already buggy :)
If we need to consider the case,
```js
var s = scaling ? (scaling.x * scaling.y < 0 ?
                   Math.abs(scaling.x) + Math.abs(scaling.y) : scaling.x + scaling.y) : 0;
return (Math.abs(s) < Numerical.EPSILON ? Numerical.EPSILON : s) / 2;
```
is better.

**Comment by sasensi** [comment]
Yes, there is the problem of the 0 value and also infinity / -infinity cases.
I did a PR #1558 concerning a related problem when setting zoom.

**Comment by lehni** [comment]
Yes I'm aware of #1558. It would solve this case here too then?

**Comment by sasensi** [comment]
No, both problems have different solutions but I mentioned it because I think that is the same kind of decision to make about how we handle null or infinite values that could crash the script.
The bug reported here have to be fixed since doing `view.scale(1,-1)` is a valid call.
But I wonder about `view.scale(0)` and `view.setZoom(0)` which seems more like invalid values.
Should we refuse these invalid values by throwing explicit error or should we approximate them as @sapics  proposed in his last comment and as I did in #1558 ?
